### DISCUSSION TITLE
Manually setting SCE optimization parameters & CPU usage of mHM ?

### DISCUSSION THREAD

**Discussion by MalveHeinz** [mhm-ufz/mHM#103] [Q&A] [answered]
Hi!

I want to optimize mHM using the shuffled complex evolution approach (SCE). When I run it, i get this error message:

  Run mHM optimization
  Start optimization
    Use SCE
STOP mo_sce: npg has to be at least 3

When i manually set the parameters  sce_nps and sce_npg to -9 in my mhm.nml file (which, according to the source code, should be the default value), it runs without any problem. My question is: is it ok to set these parameters manually or does this error indicate that something is off with my setup ?

And my next question is: is somewhere hardcoded how many CPUs mHM will use during optimization? 

Thanks a lot for your help,
Malve

**Comment by StephanThober** [comment, marked as answer]
Hi Malve,

Setting the npg manually should be fine, but there should be at least 3 points per complex. Complexes in SCE cannot be lines or points (hence the error message). However, the default value provides a reasonable choice that we also often use.

The number of CPUs used during running mHM (including the application during optimization) is not hard-coded. You can set the number of OPEN_MP threads to be used in the command line before you execute mHM via

export OMP_NUM_THREADS=#number_of_cores#

Make sure that you actually compiled mHM with the OPEN_MP switch turned on. If you use CMAKE for installation, then you can use this script for compilation https://git.ufz.de/mhm/mhm/-/blob/develop/CI-scripts/compile_OpenMP

Best,
Stephan

**Comment by 9202014710** [comment]
Can you tell me how to do the calibration? Thanks a lot.
